x-pack/plugins/observability_solution/apm/public/components/app/service_inventory/service_list/column_header_with_tooltip.tsx

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { EuiFlexGroup, EuiPopover, EuiIcon } from '@elastic/eui';
+import React, { useCallback } from 'react';
+import useToggle from 'react-use/lib/useToggle';
+
+export function ColumnHeaderWithTooltip({
+  label,
+  tooltipContent,
+}: {
+  label: string;
+  tooltipContent: string;
+}) {
+  const [isPopoverOpen, togglePopover] = useToggle(false);
+
+  const onButtonClick = useCallback(
+    (e: React.MouseEvent<HTMLButtonElement>) => {
+      e.preventDefault();
+      e.stopPropagation();
+      togglePopover();
+    },
+    [togglePopover]
+  );
+
+  return (
+    <EuiFlexGroup gutterSize="xs" alignItems="center">
+      <div>{label}</div>
+      <EuiPopover
+        panelPaddingSize="s"
+        button={
+          <button
```

Comment:
Is there a reason not to use `EuiButtonIcon` in place of a native `button` element wrapping an `EuiIcon`?

We can use the `onClick` handler and update a React Dispatch (from `useState`) without the need to cancel the default behavior in the callback defined above? This is more a curious question than saying we should do this; this deviates a bit from a lot of the popover implementations I've seen other places.

Reply:
There was an issue with the keyboard navigation support so this one was changed here: https://github.com/elastic/kibana/pull/175090/files#r1457531279  ([issue](https://github.com/elastic/kibana/issues/172561))

Reply:
I am ++ to the idea that the EUI team should be notified then, this seems like something they would care a lot about.

Reply:
True but the thing is that most of the cases with the buttons use this "jumping" behavior on hover/click.

https://github.com/elastic/kibana/assets/14139027/128dea00-2c81-424e-944a-457b3d7f16ec

 we want to avoid such a thing in the case of those buttons - maybe an extra prop to handle that in the EUI button can help for example.

Reply:
> maybe an extra prop to handle that in the EUI button can help for example

I suggested this but I suspected (and they confirmed) that this isn't necessarily behavior they want to expose from an API perspective as it's so rare that that behavior is actually valid and desired.

The override they recommend looks like:

```typescript
<EuiButtonIcon css={{ transform: 'none !important' }} />
```

Reply:
Thank you for suggesting that and sharing a solution! Unfortunately, it's not just the "jumping" behavior that we want to change: the icon shouldn't have a background (also on hover), and should have the same styles as the text next to it - basically, we want a simple icon with clicking action (not all the button styles). I am sharing what I tried and couldn't find a clean way to achieve what we currently have (the color change is tricky and I already added 9 lines of css and it's still not the same): 

https://github.com/elastic/kibana/assets/14139027/b80b46f6-d91f-4bb8-9c51-1f1644a80165

IMO we should find a cleaner way to do that rather than adding more than 10 lines of overrides at the end.
I think we can merge that PR and probably continue the discussion with eui team in a new issue where we can change that everywhere in the code at once, wdyt?

Reply:
I added a new issue [#179328](https://github.com/elastic/kibana/issues/179328) for this change